pkgs/package-overrides.nix

```diff
@@ -43,6 +43,35 @@ in
   };
 
   extensions = prev.extensions // {
+
+    mcrypt =
+      let
+        version = "0.5.6.40";
+        revision = "c37265eacdd0186cb3b0bfeb0e0104c8563807ef";
+      in
+        if lib.versionOlder prev.php.version "7.0" then
+          prev.mkExtension {
+            name = "mcrypt";
+            version = version;
+            src = pkgs.fetchFromGitHub {
+              owner = "php";
+              repo = "php-src";
+              rev = revision;
+              sha256 = "sha256-aDiGbOqy2L8Qqd6IliEN19Oa8JO1uDSEju6uUN4b0XI=";
+            } + "/ext/mcrypt";
```

Comment:
Why not use `prev.php.src` pulled in by `mkExtension` by default?

```suggestion
```

Reply:
I replaced it with prev.php.src in next commit

Reply:
Here are examples of what to pass to `mkExtension`: https://github.com/NixOS/nixpkgs/blob/3f01f05941eec3a87bbf5baae8edaea6ce3615fa/pkgs/top-level/php-packages.nix#L360

Note that `src` and build directory are already properly set based on `name`:

https://github.com/NixOS/nixpkgs/blob/3f01f05941eec3a87bbf5baae8edaea6ce3615fa/pkgs/top-level/php-packages.nix#L129

Reply:
Cleaned it again. Thanks for the references, I didn't know exactly how it handle that.